lib/src/components/NxPolicyViolationIndicator/__tests__/NxPolicyViolationIndicator.test.tsx

```diff
@@ -4,47 +4,89 @@
-    expect(getShallowComponent({ policyThreatLevel: 9 }))
-        .not.toMatchSelector('.nx-policy-violation-indicator--unspecified');
+  it('applies a ref to the top level element', function() {
+    const ref = React.createRef<HTMLDivElement>(),
+        el = renderEl({ref: ref})!;
+
+    expect(ref.current).toBe(el);
+  });
+
+  it('sets the text according to threatLevelCategory if no text is provided', function() {
+    const { container, rerender } = render(
+      <NxPolicyViolationIndicator threatLevelCategory='none'></NxPolicyViolationIndicator>
+    );
+    expect(container).toHaveTextContent('none');
+
+    rerender(<NxPolicyViolationIndicator threatLevelCategory='moderate'></NxPolicyViolationIndicator>);
+    expect(container).toHaveTextContent('moderate');
   });
 
-  it('sets the modifier class using the threatLevelCategory if both props are provided', function() {
-    const component = getShallowComponent({ policyThreatLevel: 9, threatLevelCategory: 'low' });
+  it('overrides the default threatLevelCategory text if text is provided', function() {
+    const { container } = render(
+      <NxPolicyViolationIndicator threatLevelCategory='low'>Foo</NxPolicyViolationIndicator>
+    );
 
-    expect(component).toMatchSelector('.nx-policy-violation-indicator--low');
-    expect(component).not.toMatchSelector('.nx-policy-violation-indicator--critical');
-    expect(component).not.toMatchSelector('.nx-policy-violation-indicator--unspecified');
+    expect(container).not.toHaveTextContent('low');
+    expect(container).toHaveTextContent('Foo');
   });
 
-  it('adds aria attrs to help the icon show up for screen readers', function() {
-    const component = getShallowComponent({ policyThreatLevel: 9, threatLevelCategory: 'low' });
+  it('sets threatLevelCategory to unspecified if not defined', function() {
+    const unspecifiedEl = renderEl()!;
+    expect(unspecifiedEl).toHaveTextContent('unspecified');
+  });
+
+  it('sets the threatLevelCategory by converting policyThreatLevel if it is provided',
+      function() {
+        expect(renderEl({ policyThreatLevel: 0})!).toHaveTextContent('none');
+        expect(renderEl({ policyThreatLevel: 1})!).toHaveTextContent('low');
+        expect(renderEl({ policyThreatLevel: 3})!).toHaveTextContent('moderate');
+        expect(renderEl({ policyThreatLevel: 5})!).toHaveTextContent('severe');
+        expect(renderEl({ policyThreatLevel: 9})!).toHaveTextContent('critical');
+        expect(renderEl({ policyThreatLevel: 0})!).toHaveTextContent('none');
+        expect(renderEl()!).toHaveTextContent('unspecified');
+      });
+
+  it('renders an indicator according to threatLevelCategory over policyThreatNumber of both are provided', function() {
+    expect(renderEl({ policyThreatLevel: 9, threatLevelCategory: 'low'})).toHaveTextContent('low');
+  });
+
+  it('adds aria-label to icon', function() {
```

Comment:
`aria-label` is an impl detail, the accessible name is the external behavior we want to test.  Use `toHaveAccessibleName`.

Reply:
> `aria-label` is an impl detail, the accessible name is the external behavior we want to test. Use `toHaveAccessibleName`.
thanks, updated here: [c244209](https://github.com/sonatype/sonatype-react-shared-components/pull/911/commits/c24420967b560540051227a848bf623fe0b607ba)